packages/teleport/src/Nodes/AddNode/Iam/Iam.tsx

```diff
@@ -0,0 +1,267 @@
+import React from 'react';
+import styled from 'styled-components';
+import { DialogContent, DialogFooter } from 'design/Dialog';
+import {
+  Text,
+  Box,
+  ButtonPrimary,
+  Link,
+  Alert,
+  ButtonLink,
+  ButtonSecondary,
+} from 'design';
+import TextSelectCopy from 'teleport/components/TextSelectCopy';
+import FieldInput from 'shared/components/FieldInput';
+import Validation, { Validator } from 'shared/components/Validation';
+import { Attempt } from 'shared/hooks/useAttemptNext';
+import { Rule } from 'teleport/services/joinToken';
+import DownloadLinks from 'teleport/components/DownloadLinks';
+import { Pencil } from 'design/Icon';
+
+export default function Iam({
+  token,
+  expiry,
+  attempt,
+  onGenerate,
+  isEnterprise,
+  version,
+  onClose,
+}: Props) {
+  const { hostname, port } = window.document.location;
+  const host = `${hostname}:${port || '443'}`;
+
+  const [rules, setRules] = React.useState<RuleForm[]>([
+    {
+      awsAccount: '',
+      awsArn: '',
+      isCollapsed: false,
+    },
+  ]);
+
+  function setRuleAtIndex(index: number, rule: RuleForm) {
+    const newRules = [...rules];
+    newRules[index] = rule;
+    setRules(newRules);
+  }
+
+  function handleRemove(index: number) {
+    const newRules = [...rules];
+    newRules.splice(index, 1);
+    setRules(newRules);
+  }
+
+  function handleGenerate(validator: Validator) {
+    if (!validator.validate()) {
```

Comment:
What's going on here? `validator` doesn't appear to actually be used after this check.

Reply:
`validator.validate()` will run all methods in the form's `FieldInput` `rule` fields and display and show the error message in case it fails.

That's why we can just return in this case, the validator will take care of the rest. The same is used in multiple places

Reply:
Can you add a comment to this effect in the code as it's not clear. Be as detailed as you need to be :)

Reply:
Added a comment there, thanks